src/items/deleteItemDialog.component.test.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { screen, RenderResult, waitFor, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import DeleteItemDialog, {
+  DeleteItemDialogProps,
+} from './deleteItemDialog.component';
+import { getItemById, renderComponentWithBrowserRouter } from '../setupTests';
+import { Item } from '../app.types';
+
+describe('delete item dialog', () => {
+  let props: DeleteItemDialogProps;
+  let user;
+  const onClose = jest.fn();
+  const onChangeItem = jest.fn();
+  let item: Item | undefined;
+
+  const createView = (): RenderResult => {
+    return renderComponentWithBrowserRouter(<DeleteItemDialog {...props} />);
+  };
+
+  beforeEach(() => {
+    item = getItemById('KvT2Ox7n');
+    props = {
+      open: true,
+      onClose: onClose,
+      item: item,
+      onChangeItem: onChangeItem,
+    };
+    user = userEvent; // Assigning userEvent to 'user'
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it('renders correctly', async () => {
+    createView();
+    let baseElement;
+    await act(async () => {
+      baseElement = createView().baseElement;
+    });
+    expect(baseElement).toMatchSnapshot();
```

Comment:
Why use snapshots in this case? We managed to avoid them for the others

Reply:
The other delete dialogs should have doing snapshots tests as the text is broken up into different elements. The snapshot tests the rendering of dialog better.

Reply:
I think if they are simple its not necessarily worth it, as it is possible to overcome e.g. ![image](https://github.com/ral-facilities/inventory-management-system/assets/90245114/18272621-b869-4ac0-b588-fdb2dcaef9a4) and I thought they were generally considered a last resort as they are so picky, I am alright with leaving it in for this one though.